solc/CommandLineInterface.cpp

```diff
@@ -1164,57 +1168,56 @@ void CommandLineInterface::outputCompilationResults()
 	// do we need AST output?
 	handleAst();
 
-	if (
-		!m_compiler->compilationSuccessful() &&
-		m_options.output.stopAfter == CompilerStack::State::CompilationSuccessful
-	)
-	{
-		serr() << endl << "Compilation halted after AST generation due to errors." << endl;
-		return;
-	}
-
-	vector<string> contracts = m_compiler->contractNames();
-	for (string const& contract: contracts)
+	CompilerOutputs astOutputSelection;
+	astOutputSelection.astCompactJson = true;
+	if (m_options.compiler.outputs != CompilerOutputs() && m_options.compiler.outputs != astOutputSelection)
```

Comment:
if "the selected compiler outputs are different from no selected compiler outputs (if any option was set)" AND "the selected compiler outputs are different from the ones where only `astCompactJson` was selected (if any option except `astOutputSelection` was set.." - isn't it enough to just check for the second part? but I'm not really sure that I fully understand this..

Reply:
This `if` is meant to run only if you select at least one output other than `--ast-compact-json`. Otherwise there is either nothing to output or the output has already been handled by `handleAst()` above.